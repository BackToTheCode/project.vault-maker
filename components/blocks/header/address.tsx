/** @jsx jsx */
import { jsx } from '@emotion/core';
import React from 'react';
import { Flex, Box, Text } from 'rebass';

export interface IAddressProps {
  variant?: string;
  children?: React.ReactNode;
  address?: string;
  ern?: string;
  connected?: boolean;
}

const styles = {
  container: {
    width: '100%',
    justifyContent: 'flex-end',
    mr: 20,
    alignItems: 'center'
  },
  indicator: {
    borderRadius: '50%',
    position: 'relative',
    width: '8px',
    height: '8px',
    right: 6,
    top: 6
  },
  addressContainer: {
    px: 4,
    py: 3,
    bg: 'lightGrey',
    borderRadius: '4px'
  }
};

const Address: React.FC<IAddressProps> = ({ connected, address, ern }) => {

  const variant = connected ? 'indicator.connected' : 'indicator.unconnected';

  return (
    <Flex width={1} sx={styles.container}>
      <Box
        sx={{ textAlign: 'left', flexDirection: 'column', maxWidth: '130px' }}
      >
        <Box variant={variant} sx={styles.indicator} />
        <Box sx={styles.addressContainer}>
          <Text
            sx={{ overflow: 'hidden', textOverflow: 'ellipsis' }}
            variant={'body.regular'}
          >
            {address ? address : 'Not connected'}
          </Text>
        </Box>
        {ern && (
          <Box sx={{ marginTop: 2 }}>
            <Text variant={'body.small'} sx={{ color: 'grey' }}>
              {ern}
            </Text>
          </Box>
        )}
      </Box>
    </Flex>
  );
};

export default Address;
