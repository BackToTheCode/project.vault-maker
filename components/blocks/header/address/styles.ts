const styles = {
  address: {
    color: 'rgb(255,255,255)',
    display: ['none', 'block', 'block'],
    maxWidth: '140px',
    overflow: 'hidden',
    px: 4,
    py: 4,
    textOverflow: 'ellipsis'
  },
  container: {
    mr: 20,
    position: 'relative',
    textAlign: 'left',   
  },
  indicator: {
    borderRadius: '50%',
    bottom: [0, '16px', '16px'],
    connected: (isConnected: boolean): any => (isConnected ? { bg: 'primary' } : { bg: 'warning' }),
    height: '8px',
    left: '-15px',
    position: 'absolute',
    width: '8px',
  }
};

export default styles;
