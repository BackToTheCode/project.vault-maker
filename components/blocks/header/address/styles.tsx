const styles = {
  container: {
    width: '100%',
    justifyContent: 'flex-end',
    mr: 20,
    alignItems: 'center'
  },
  innerContainer: {
    textAlign: 'left',
    flexDirection: 'column',
    maxWidth: '140px',
    position: 'relative',
    top: '-5px'
  },
  indicator: {
    borderRadius: '50%',
    position: 'relative',
    width: '8px',
    height: '8px',
    right: 7,
    top: '22px'
  },
  addressContainer: {
    px: 4,
    py: 4,
    bg: 'lightGrey',
    borderRadius: '4px',
  },
  ernContainer: {
    marginTop: 3,
    fontWeight: 'bold',
    textTransform: 'uppercase'
  },
  ern: { color: 'grey' }
};

export default styles;