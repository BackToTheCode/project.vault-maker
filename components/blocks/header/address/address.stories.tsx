import { checkA11y } from '@storybook/addon-a11y';
import { boolean, withKnobs } from '@storybook/addon-knobs';
import React from 'react';
import { Box } from 'rebass';
import { Address } from '../address';

const Background = ({ children }) => (
  <Box sx={{ bg: 'rgb(55,74,97)', ml: 7}}>{children}</Box>
);

export default {
  component: Address,
  decorators: [checkA11y],
  parameters: {
    backgrounds: [
      { name: 'midGrey', value: 'rgb(55,74,97)', default: true },
      { name: 'primary', value: 'rgb(26, 171, 155)' }
    ]
  },
  title: 'Elements|Address'
};

export const Unconnected = () => (
  <Background>
    <Address />
  </Background>
);

export const Connected = () => (
  <Background>
    <Address
      isConnected={true}
      address="0x61049F5e03Bfe3823f274C479158A94bcA26456c"
    />
  </Background>
);

export const WithKnobs = () => (
  <Background>
    <Address
      isConnected={boolean('isConnected', false)}
      address="0x61049F5e03Bfe3823f274C479158A94bcA26456c"
    />
  </Background>
);

WithKnobs.story = {
  decorators: [withKnobs]
};
