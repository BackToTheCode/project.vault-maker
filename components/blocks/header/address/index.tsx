/** @jsx jsx */
import { jsx } from '@emotion/core';
import React, { FC } from 'react';
import { Box, Flex, Text } from 'rebass';
import styles from './styles';

export interface AddressProps {
  variant?: string;
  children?: React.ReactNode;
  address?: string;
  ern?: string;
  isConnected?: boolean;
}

export const Address: FC<AddressProps> = ({ isConnected, address, ern }) => {
  const variant = isConnected ? 'indicator.connected' : 'indicator.unconnected';

  const connectedMessage = isConnected
    ? address
      ? address
      : 'Connected'
    : 'Not connected';

  return (
    <Flex sx={styles.container}>
      {/* <Box sx={styles.innerContainer}> */}
        <Box variant={variant} sx={styles.indicator} />
        <Box sx={styles.addressContainer}>
          <Text
            sx={styles.address}
            variant={'body.regular'}
          >
            {connectedMessage}
          </Text>
        </Box>
        {ern && (
          <Box sx={styles.ernContainer}>
            <Text variant={'body.small'} sx={styles.ern}>
              {ern}
            </Text>
          </Box>
        )}
      {/* </Box> */}
    </Flex>
  );
};
Address.displayName = 'Address';


