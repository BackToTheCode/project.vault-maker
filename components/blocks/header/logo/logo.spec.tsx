import { render } from '@testing-library/react';
import React from 'react';
import { Address } from '../address';


describe('<Logo /> spec', () => {
  test('it renders the component', () => {
    // Arrange
    const expectedText = 'Not connected';

    // Act
    const { getByText } = render(<Address />);

    // Assert
    expect(getByText(expectedText)).toMatchSnapshot();
  });
});
