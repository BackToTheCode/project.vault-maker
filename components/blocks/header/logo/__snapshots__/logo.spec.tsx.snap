// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> spec it renders the component 1`] = `
<div
  class="css-b300w3"
>
  Not connected
</div>
`;
